### DISCUSSION TITLE
Winget Telemetry

### DISCUSSION THREAD

**Discussion by sirjyo** [microsoft/winget-cli#3371] [Q&A]
What kind of diagnostic data does WinGet send based on the telemetry levels?

**Comment by denelon** [comment]
It's primarily success and failure metrics for individual commands. You are welcome to look in the source code.

We use the data to determine if product health is improving or declining. In some cases (commands related to an installer/uninstaller), the data can help us to understand if the exceptions thrown are due to the code paths in WinGet, or in the installer/uninstaller being invoked.
